_posts/2020-06-13-IrishGen-RDF-Linked-Data-Semantic-Web.md

```diff
@@ -0,0 +1,354 @@
+database, it can be very frustrating and confidence draining.
+
+There is not much that can be done for this kind of problem overall.
+More careful curators and automation can decrease the occurance as
+discussed in "Human Curation and Digital Datasets" but this will never
+fully elimiate the possibility.
+
+Another form of error can cause real problems in a system.  Take the
+below more fleshed out version:
+
+```turtle
+@base <http://example.com/LL/dal_corpri_arad.trig> .
+@prefix foaf: <http://xmlns.com/foaf/0.1/> .
+@prefix rel: <http://purl.org/vocab/relationship/> .
+
+<http://example.com/LL> {
+  <#Flaithbertach> a foaf:Person ;
+                   rel:childOf <#Crunmael>.
+}
+```
+
+The above is a full TRiG file that will be parsed successfully by a
+system that can parse the TRiG file format.  The file above translated
+into English states: "In LL, Flaithbertach is a person and is the
+child of Crunmael".  As the astute reader will have noticed, Crunmael
+is nowhere mentioned.  Due to the Open World Assumption, the existence
+of Crumael will be assumed by the system.  In fact, in the face of
+reasoning and the `rel` ontology, he will also be assumed to be a
+person.  While this can be useful, it can also create "ghost
+individuals" who exist only because a human made a mistake somewhere.
+Additionally, if there are two individuals with the same name and thus
+the same URL, a completely skewed graph can result because the
+Triplestore will assume they are referring to the same person.  The
+curators avoid this siutation generally by appending a random fragment
+of a universally unique identifier (UUID, for a formal definition, see
+[RFC 4122](https://www.ietf.org/rfc/rfc4122.txt)) to the URL to
+distinguish between two individuals with the same name.  However,
+occationally the UUID fragment will be missed during the creation of
+the URL and two seperate people will be accidentally merged together,
+which is the inverse of the "ghost person" problem.
+
+The most difficult and insidious form of error that can occur is the
+`owl:sameAs` error.  One of the most useful things IrishGen can do is
+to not only extract information and make it searchable but also
+connect the same individuals and lines of descent across MSS.  The way
+that IrishGen does this is by using the OWL 2 `sameAs` built-in
+predicate.  This predicate is special as it has properties that others
+do not (for a formal description of `owl:sameAs` as it pertains to its
+special properties see [OWL 2 Web Ontology Language Profiles (Second
+Edition)](https://www.w3.org/TR/2012/REC-owl2-profiles-20121211/)).
+Suffice it to say, it is a powerful predicate and can cause problems
+of its own.  If an individual is mistakenly marked as `owl:sameAs`
+another unrelated individual, this will cause a Triplestore to return
+an incorrect graph and thus either confuse, in the best case, or
+mislead, in the worst, a user.  This comes down to the curators and
```

Comment:
Is it worth mentioning at this point that triplestores will often not make it apparent why it has returned the graph it has, meaning that this sort of error can be difficult to track down? This will presumably be getting its own blogpost at some point, of course.